src/main/java/de/tum/cit/aet/artemis/core/web/admin/AdminBuildJobQueueResource.java

```diff
@@ -95,8 +95,11 @@ public ResponseEntity<List<BuildAgentInformation>> getBuildAgentSummary() {
     @GetMapping("build-agent")
     public ResponseEntity<BuildAgentInformation> getBuildAgentDetails(@RequestParam String agentName) {
         log.debug("REST request to get information on build agent {}", agentName);
-        BuildAgentInformation buildAgentDetails = localCIBuildJobQueueService.getBuildAgentInformation().stream().filter(agent -> agent.name().equals(agentName)).findFirst()
-                .orElse(null);
+        BuildAgentInformation buildAgentDetails = localCIBuildJobQueueService.getBuildAgentInformation().stream().filter(agent -> agent.buildAgent().name().equals(agentName))
+                .findFirst().orElse(null);
+        if (buildAgentDetails == null) {
+            return ResponseEntity.notFound().build();
+        }
         return ResponseEntity.ok(buildAgentDetails);
```

Comment:
Why not use the optional directly?
```suggestion
        Optional<BuildAgentInformation> buildAgentDetails = localCIBuildJobQueueService.getBuildAgentInformation().stream().filter(agent -> agent.buildAgent().name().equals(agentName))
                .findFirst();
        if (buildAgentDetails.isEmpty()) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.ok(buildAgentDetails.get());
```